backend/data/helpers/recipeScraper.js

```diff
@@ -1,39 +1,109 @@
 
 const rp = require('request-promise');
 const $ = require('cheerio');
+const numericQuantity = require("numeric-quantity");
 
-checkUrl = (newRecipeObj) =>{
-  console.log(newRecipeObj);
-  const checkRecipeUrl = newRecipeObj.recipeUrl;
-  if(checkRecipeUrl.indexOf('allrecipes.com')){
-      allRecipeScraper(checkRecipeUrl);
-  }else if(checkRecipeUrl.indexOf('pinchofyum.com')){
-      pinchOfYumScraper(checkRecipeUrl);
+checkUrl = async (newRecipeObj) => {
+
+  const checkRecipeUrl = newRecipeObj.link;
+  if(checkRecipeUrl.indexOf('allrecipes.com') !== -1 ){
+      await allRecipeScraper(checkRecipeUrl);
+  }else if(checkRecipeUrl.indexOf('pinchofyum.com') !== -1 ){
+      await pinchOfYumScraper(checkRecipeUrl);
   }else return{
     checkRecipeUrl
   }
   return newRecipeObj;
 }
-allRecipeScraper = url =>{
-    rp(url)
+
+/* Parse the ingredient array & split it into objects */
+const parseIngredients = (ingArr) => {
+  let ingredientArray = [];
+
+  ingArr.forEach( (line) => {
+    // Split each line into an array
+    let lineArr = line.split(" ");
+
+    // Set up the empty object & needed variables
+    let tempObj = { amount: null, measurement: null, name: ""};
+    let nameArr = lineArr;
+
+    // Scan through each line's array values for needed components
+    for( let i = 0; i < lineArr.length; i++ ) {
+
+      // Check if this value is a number
+      let testNum = Number(lineArr[i]);
+
+      // Fraction?
+      if( lineArr[i].indexOf("/") !== -1 ){
+        testNum = numericQuantity(lineArr[i]) > -1 ? numericQuantity(lineArr[i]) : testNum;
+      }
+
+      if( !Number.isNaN(testNum) ){
+        nameArr[i] = '';
+        tempObj.amount = testNum;
+      }
+
+      // Look for measurement - this needs a better method
+      const msmts = ['c','cup','cups','T','tbsp','tablespoon','tablespoons','tsp','teaspoon','teaspoons','oz','ounce','ounces','stalk','stalks','pinch','pinches','dash','dashes','pound','pounds','handful','handfulls','pint','pints','qt','quart','quarts','lb','pound','pounds','gallon','gallons','liter','liters'];
+
+      if( msmts.indexOf(lineArr[i]) !== -1 ){
+        tempObj.measurement = lineArr[i];
+        nameArr[i] = '';
+      }
+
+      // Join the remaining nameArr to form the name
+      tempObj.name = nameArr.join(' ').trim();
+
+    }
+    ingredientArray.push(tempObj);
+  });
+
+  return(ingredientArray);
+};
+
+allRecipeScraper = async (url) => {
+  rp(url)
   .then(function(html){
-    let name, image, link, ingredients, directions;
-    const allRecipeJSON = {name: '', image: '', link: '', ingredients: '', directions: ''} 
+    //let name, image, link, ingredients, directions;
+    const allRecipeJSON = {name: '', image: '', link: '', ingredients: '', directions: '', } 
```

Comment:
Don't need the trailing comma
```suggestion
    const allRecipeJSON = {name: '', image: '', link: '', ingredients: '', directions: ''} 
```